src/jvmMain/kotlin/io/github/oshai/kotlinlogging/internal/KLoggerFactory.kt

```diff
@@ -12,10 +14,16 @@ internal actual object KLoggerFactory {
   internal actual fun logger(name: String): KLogger {
     if (System.getProperty("kotlin-logging-to-jul") != null) {
       return JulLoggerFactory.wrapJLogger(JulLoggerFactory.jLogger(name))
-    } else if (System.getProperty("kotlin-logging-to-logback") != null) {
-      return LogbackLoggerFactory.wrapLogbackLogger(LogbackLoggerFactory.logbackLogger(name))
     }
-    // default to slf4j
-    return Slf4jLoggerFactory.wrapJLogger(Slf4jLoggerFactory.jLogger(name))
+    // Try Logback and if that fails then default to plain SLF4J
+    return if (LoggerFactory.getILoggerFactory().isLogback()) {
+      LogbackLoggerFactory.logbackLogger(name)
+    } else {
+      Slf4jLoggerFactory.wrapJLogger(Slf4jLoggerFactory.jLogger(name))
+    }
   }
 }
+
+private fun ILoggerFactory.isLogback() =
+  System.getProperty("kotlin-logging-to-logback") == "true" &&
```

Comment:
I think it's better to have the property check outside, as it does not depend on class, and there are other property checks (jul) on the caller.

Also I think the extension method gives a fluent api, but otoh it doesn't really use the public properties of the subject, so it's less clear. If this will be inside the class, is it possible to check isInstance instead? - I mean in such case it should allow to import that class, isn't it?

Reply:
Explanation: originally I implemented automatically enabling `LogbackLoggerFactory` when Logback is in use and then added the opt-in check for system property -- thus the extra complexity.
I now simplified the code again.